quickstart/201-k8s-cluster-with-aks-applicationgateway-ingress/main.tf

```diff
@@ -116,47 +154,39 @@ resource "azurerm_application_gateway" "network" {
 
   request_routing_rule {
     name                       = local.request_routing_rule_name
+    priority                   = 1
     rule_type                  = "Basic"
     http_listener_name         = local.listener_name
     backend_address_pool_name  = local.backend_address_pool_name
     backend_http_settings_name = local.http_setting_name
-    priority                   = 1
   }
-}
-
-resource "azurerm_kubernetes_cluster" "k8s" {
-  name       = var.aks_cluster_name
-  location   = azurerm_resource_group.rg.location
-  dns_prefix = var.aks_dns_prefix
-
-  identity {
-    type = "SystemAssigned"
+  lifecycle {
```

Comment:
I think I see why that was added. We should add a comment that explains why this lifecycle ignore changes is required for this scenario.

Reply:
@grayzu @lonegunmanb 

From my understanding of a GitHub Issue on this topic [1], since this sample is creating an Application Gateway, which is later managed by an Ingress Controller, there is no need to create a backend address pool (BEP). However, the BEP is still required by the resource. Therefore, "lifecycle:ignore_changes" is used to prevent TF from managing the gateway.

[1] https://github.com/hashicorp/terraform-provider-azurerm/issues/10009

Reply:
@grayzu I added the requested comment.